src/precice/config/ParticipantConfiguration.cpp

```diff
@@ -265,17 +257,16 @@ void ParticipantConfiguration::xmlTagCallback(
     impl::PtrParticipant p(new impl::ParticipantState(name, _meshConfig));
     _participants.push_back(p);
   } else if (tag.getName() == TAG_PROVIDE_MESH) {
-    PRECICE_ASSERT(_dimensions != 0); // setDimensions() has been called
```

Comment:
We now set the dimensions later. I am not sure if this can have any side-effects.